source/renderer/util/editor-tabs.js

```diff
@@ -200,7 +203,12 @@ module.exports = class EditorTabs {
 
     // If given, call the callback
     if (this._intentCallback) {
-      this._intentCallback(hash, (closeIntent) ? 'close' : 'select')
+      // middle mouse click closes tab
+      if (event.type === 'auxclick' && event.button === 1) {
+        this._intentCallback(hash, 'close')
+      } else {
+        this._intentCallback(hash, (closeIntent) ? 'close' : 'select')
```

Comment:
Instead of this if/else-switch, we can make it slightly more readable, by assigning `let middleClick` (or something similar) the result of the if condition and then in the ternary if/else query for `closeIntent || middleClick`.

Reply:
and done.